internal/fs/inode/dir.go

```diff
@@ -771,15 +771,19 @@ func (d *dirInode) CreateChildFile(ctx context.Context, name string) (*Core, err
 	}, nil
 }
 
-func (d *dirInode) CreateLocalChildFile(name string) (*Core, error) {
+func (d *dirInode) CreateLocalChildFile(name string, updateInodeFn func(core Core) error) error {
```

Comment:
Since we are modifying the d.cache, that means this method requires LOCKS_REQUIRED(d).